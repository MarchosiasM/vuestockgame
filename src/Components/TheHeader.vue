<template>
    <nav class="navbar navbar-expand-md navbar-light bg-light">
      <router-link to="/" class="navbar-brand"><a>
        Stocks App
        </a></router-link>
      <button class="navbar-toggler" type="button" data-toggle="collapse"
        data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false"
        aria-label="Toggle navigation">
          <span class="navbar-toggler-icon"></span>
        </button>
      <div class="collapse navbar-collapse" id="navbarNav">
        <ul class="navbar-nav">
            <router-link to="/stocks" activeClass="active" tag="li">
            <a class="nav-link">
              Stocks
            </a>
            </router-link>
            <router-link to="/portfolio" activeClass="active" tag="li">
            <a class="nav-link">
              Portfolio
            </a>
            </router-link>
        </ul>
        <ul class="navbar-nav ml-auto">
          <li class="nav-item ml-auto">
            <a class="nav-link" @click="endDay">
              End Day
            </a>
          </li>
          <li class="nav-item">
            <a class="nav-link">
              Save & Load
            </a>
          </li>
          <li class="nav-item">
            <a class="nav-link">
              Funds: {{ funds }}
            </a>
          </li>
        </ul>
      </div>
    </nav>
</template>

<script>
export default {
  data() {
    return {};
  },
  computed: {
    funds() {
      return this.$store.getters.funds;
    }
  },
  methods: {
    endDay() {
      this.$store.dispatch('nextDay');
    }
  }
};
</script>
