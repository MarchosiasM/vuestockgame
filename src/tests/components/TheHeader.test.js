import { mount } from '@vue/test-utils';
import TheHeader from '../../Components/TheHeader.vue';

describe('Component', () => {
  const wrapper = mount(TheHeader);
  it('renders properly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('initializes with funds equal to 10,000', () => {
    expect(wrapper.vm.funds).toEqual(10000);
  })
});
