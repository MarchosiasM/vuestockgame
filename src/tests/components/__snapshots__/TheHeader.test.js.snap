// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders properly 1`] = `
<nav
  class="navbar navbar-expand-md navbar-light bg-light"
>
  <router-link-stub
    class="navbar-brand"
    to="/"
  >
    <a>
      
    Stocks App
    
    </a>
  </router-link-stub>
   
  <button
    aria-controls="navbarNav"
    aria-expanded="false"
    aria-label="Toggle navigation"
    class="navbar-toggler"
    data-target="#navbarNav"
    data-toggle="collapse"
    type="button"
  >
    <span
      class="navbar-toggler-icon"
    />
  </button>
   
  <div
    class="collapse navbar-collapse"
    id="navbarNav"
  >
    <ul
      class="navbar-nav"
    >
      <router-link-stub
        activeclass="active"
        tag="li"
        to="/stocks"
      >
        <a
          class="nav-link"
        >
          
          Stocks
        
        </a>
      </router-link-stub>
       
      <router-link-stub
        activeclass="active"
        tag="li"
        to="/portfolio"
      >
        <a
          class="nav-link"
        >
          
          Portfolio
        
        </a>
      </router-link-stub>
    </ul>
     
    <ul
      class="navbar-nav ml-auto"
    >
      <li
        class="nav-item ml-auto"
      >
        <a
          class="nav-link"
        >
          
          End Day
        
        </a>
      </li>
       
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
        >
          
          Save & Load
        
        </a>
      </li>
       
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
        >
          
          Funds: 10000
        
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
